lib/matplotlib/axes/_base.py

```diff
@@ -2289,41 +2289,45 @@ def _update_line_limits(self, line):
         if path.vertices.size == 0:
             return
 
-        line_trans = line.get_transform()
+        line_trf = line.get_transform()
 
-        if line_trans == self.transData:
+        if line_trf == self.transData:
             data_path = path
-
-        elif any(line_trans.contains_branch_seperately(self.transData)):
-            # identify the transform to go from line's coordinates
-            # to data coordinates
-            trans_to_data = line_trans - self.transData
-
-            # if transData is affine we can use the cached non-affine component
-            # of line's path. (since the non-affine part of line_trans is
-            # entirely encapsulated in trans_to_data).
+        elif any(line_trf.contains_branch_seperately(self.transData)):
+            # Compute the transform from line coordinates to data coordinates.
+            trf_to_data = line_trf - self.transData
+            # If transData is affine we can use the cached non-affine component
+            # of line's path (since the non-affine part of line_trf is
+            # entirely encapsulated in trf_to_data).
             if self.transData.is_affine:
                 line_trans_path = line._get_transformed_path()
                 na_path, _ = line_trans_path.get_transformed_path_and_affine()
-                data_path = trans_to_data.transform_path_affine(na_path)
+                data_path = trf_to_data.transform_path_affine(na_path)
             else:
-                data_path = trans_to_data.transform_path(path)
+                data_path = trf_to_data.transform_path(path)
         else:
-            # for backwards compatibility we update the dataLim with the
+            # For backwards compatibility we update the dataLim with the
             # coordinate range of the given path, even though the coordinate
             # systems are completely different. This may occur in situations
             # such as when ax.transAxes is passed through for absolute
             # positioning.
             data_path = path
 
-        if data_path.vertices.size > 0:
-            updatex, updatey = line_trans.contains_branch_seperately(
-                self.transData)
-            self.dataLim.update_from_path(data_path,
-                                          self.ignore_existing_data_limits,
-                                          updatex=updatex,
-                                          updatey=updatey)
-            self.ignore_existing_data_limits = False
+        if not data_path.vertices.size:
+            return
+
+        updatex, updatey = line_trf.contains_branch_seperately(self.transData)
+        if self.name != "rectilinear":
```

Comment:
We store the projection in a public attribute `ax.name`?  That seems ripe for abuse?  I took a quick look around and this is not used a lot.   Is this actually how we tell what the projection is?

Reply:
This is at least how Line2D.recache() (which, although slightly obscure, is very often indirectly called in practice) does the same check.  It's not clear either to me how else we could check that...